### PR TITLE
Alerting: wrap top level components in ErrorBoundary

### PR DESCRIPTION
**Pull request by domasx2**
**What this PR does / why we need it**:
Wraps top level unified alerting components in an error boundary to not crash entire grafana when something unexpected happens.

Introduces `withErrorBoundary` to be able to easily wrap components, deleted some empty file that was accidentally commited earlier

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/alerting-squad/issues/77

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/hoc/withErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { ErrorBoundaryAlert } from '@grafana/ui';
+import React, { ComponentType } from 'react';
+import { WithAlertBoxProps } from '@grafana/ui/src/components/ErrorBoundary/ErrorBoundary';
```
Comment (mckn) [inline public/app/core/hoc/withErrorBoundary.tsx:3]: This should not be imported this way. You need to export if from `grafana/ui` and import it via the regular package import.

File: packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ export class ErrorBoundary extends PureComponent<Props, State> {
   }
 }
 
-interface WithAlertBoxProps {
+export interface WithAlertBoxProps {
```
Comment (mckn) [inline packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx:49]: Probably need to add some docs here if we are going to export it.

**Review by mckn — COMMENTED** [review]
Some small nits but apart from that it looks really good. Maybe it would be handy to have the `withErrorBoundary` HOC in `grafana/ui`?

**Comment by domasx2** [comment]
> Some small nits but apart from that it looks really good. Maybe it would be handy to have the `withErrorBoundary` HOC in `grafana/ui`?

what would be involved in doing that? Do I have to update docs/ storybook/ something else?

File: packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx (hunk 2 above)
Comment (domasx2) [reply packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx]: done

File: public/app/core/hoc/withErrorBoundary.tsx (hunk 1 above)
Comment (domasx2) [reply public/app/core/hoc/withErrorBoundary.tsx]: done! Also renamed to `ErrorBoundaryProps` to be consistent with everything else

**Comment by mckn** [comment]
> > Some small nits but apart from that it looks really good. Maybe it would be handy to have the `withErrorBoundary` HOC in `grafana/ui`?
> 
> what would be involved in doing that? Do I have to update docs/ storybook/ something else?

hmm, not sure we need storybook for this one. But adding docs in form of code comments should be good.

**Comment by mckn** [comment]
> > > Some small nits but apart from that it looks really good. Maybe it would be handy to have the `withErrorBoundary` HOC in `grafana/ui`?
> > 
> > 
> > what would be involved in doing that? Do I have to update docs/ storybook/ something else?
> 
> hmm, not sure we need storybook for this one. But adding docs in form of code comments should be good.

You basically need to add a code comment on top of the export telling if the component should be considered `public`, `internal`, `alpha` or `beta`.

```
/***
*@public
***/
export interface ..

```

**Comment by domasx2** [comment]
@mckn  moved the hoc into `@grafana/ui`, begging approvals :)

File: packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -46,14 +46,19 @@ export class ErrorBoundary extends PureComponent<Props, State> {
   }
 }
 
-interface WithAlertBoxProps {
+export interface ErrorBoundaryProps {
```
Comment (mckn) [inline packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx:49]: I think we might need to add a comment with at least the `@public` attribute above this interface as well.

**Review by mckn — APPROVED** [review]
Looks really great! Added one small nit but if the CI turns green I think we can merge this.

File: packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx (hunk 3 above)
Comment (domasx2) [reply packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx]: done, thanks :)

**Pull request merged**